src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1641,9 +1664,9 @@ TService IDocumentServiceProvider.GetService<TService>()
             private sealed class SourceTextLoader : TextLoader
             {
                 private readonly SourceTextContainer _textContainer;
```

Comment:
GitHub is making this hard to navigate through, but should `_filePath` be nullable? There's explicit context that the `FilePath` can't be null above when being added. How do we get a `SourceTextLoader` associated with a null filepath?

Reply:
So this only gets used in RemoveSourceTextContainer, because we have to do a close and then a remove. That code path was just lazy and passed filePath: null because it didn't really matter.

I guess I can just pass the right thing, but practically this subtype itself doesn't really care -- if it moved elsewhere there's no reason it can't have it. The caller just doesn't need it.

I'm also not sure if I can just delete that entirely; because I wanted RemoveDocument() in general to be free-threaded, I removed the requirement for documents to be closed before removed. I'm happy to try just deleting this entirely, but that might be worth doing in a separate PR since I might need the ASP.NET Razor team to do some testing on that one.

Reply:
Thoughts?

Reply:
Please add a comment with explanation. I think it's the right approach deleting this, but understand that this may not be the right PR for it.